plugin/switch.vim

```diff
@@ -116,6 +116,12 @@ let g:switch_builtins =
       \     '(\([^()]\{-}\))\s*=>\s*{':                 'function(\1) {',
       \     '\(\k\+\)\s*=>\s*{':                        'function(\1) {',
       \   },
+      \   '
+      \   'javascript_es6_declarations': {
+      \     'var ': 'let ',
+      \     'let ': 'const ',
+      \     'const ': 'let ',
+      \   },
```

Comment:
With this format of the patterns, the key is considered a full-blown regex, which means this will also match something like `foolet bar` or `megaconst baz`. I think it would be better to restrict it like this `'\<var\>': 'let'`. The `\<` brackets match the border of a word (like `\b` in perl-style regexes, but these ones have a direction). That way, it's not necessary to keep the space after the word, either, not that it matters much. Though, you could even do something like `'\<var\s\+': 'let '`, which should transform any extra whitespace into a single space (so, if you have some messy code that has weird amounts of spacing, that would normalize it a bit).